File: main-svc/src/main/java/com/chilly/main_svc/mapper/ListToStringConverter.java
package com.chilly.main_svc.mapper;

import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;

import java.util.Arrays;
import java.util.List;

@Converter
public class ListToStringConverter implements AttributeConverter<List<String>, String> {
    private final String DELIMITER = "###";
    @Override
    public String convertToDatabaseColumn(List<String> strings) {
        if (strings == null || strings.isEmpty()) {
            return null;
        }
        return String.join(DELIMITER, strings);
    }

    @Override
    public List<String> convertToEntityAttribute(String s) {
        if (s == null) {
            return List.of();
        }
        return Arrays.asList(s.split(DELIMITER));
    }
}

File: places-svc/src/main/kotlin/com/chilly/places_svc/mapper/ListToStringConverter.kt
package com.chilly.places_svc.mapper

import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter
import java.util.Collections.emptyList

@Converter
class ListToStringConverter : AttributeConverter<List<String>?, String?> {

    override fun convertToDatabaseColumn(strings: List<String>?): String? {
        if (strings.isNullOrEmpty()) {
            return null
        }
        return strings.joinToString(separator = DELIMITER)
    }

    override fun convertToEntityAttribute(s: String?): List<String> {
        s ?: return emptyList()
        return s.split(DELIMITER.toRegex())
    }

    companion object {
        const val DELIMITER = "###"
    }
}